File: strings/commonCharacters/commonCharacters.py
# Find common characters in an array of strings
# O(n*m) time | O(m) space
def commonCharacters(strings):
    # Write your code here.
    fSet = set(strings[0])
    for i in range(1, len(strings)):
        fSet = fSet.intersection(set(strings[i]))
        
    return list(fSet)

# Example usage:
strings = ["bella", "label", "roller"]
print(commonCharacters(strings)) # Output: ['e', 'l']

File: strings/commonCharacters/commonCharacters.js
// Find common characters in an array of strings
// O(n*m) time | O(m) space
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
function commonCharacters(strings) {
    var fSet = new Set(strings[0]);
    var _loop_1 = function (i) {
        var set = new Set(strings[i]);
        fSet = new Set(__spreadArray([], fSet, true).filter(function (char) { return set.has(char); }));
    };
    for (var i = 1; i < strings.length; i++) {
        _loop_1(i);
    }
    return Array.from(fSet);
}
// Example usage:
var strings = ["bella", "label", "roller"];
console.log(commonCharacters(strings)); // Output: ['e', 'l']
